### PR TITLE
Merge volatile module, store persistent and volatile module in root dir

### PR DESCRIPTION
**Pull request by JiayuZzz**
### What is changed and how it works?

Cherry-pick volatile branch, store persistent and volatile module to independent directories.

### Check List 

Tests 

- Unit test
- Stress test
- 
Side effects

No

### CODE REVIEW

**Review by iyupeng — APPROVED** [review]
Just 2 questions:
* Will we use different namespaces for `persistent` and `volatile`, in the future? Otherwise, these 2 modules may not be compiled into one app.

* Also, will there be some common parts being extracted from both modules, e.g., header files?

**Comment by JiayuZzz** [comment]
1. Yes, this is a todo task. Also, as c API can't be isolated by namespace, we may need to do some rename work.
2. Yes, like Status and common types.

**Comment by JiayuZzz** [comment]
> Just 2 questions:
> 
> * Will we use different namespaces for `persistent` and `volatile`, in the future? Otherwise, these 2 modules may not be compiled into one app.
> * Also, will there be some common parts being extracted from both modules, e.g., header files?

Yes, this is a todo task. Also, as c API can't be isolated by namespace, we may need to do some rename work.
Yes, like Status and common types.

File: volatile/engine/data_record.hpp [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    return false;
+  }
+
+  StringView Key() const { return StringView(data, entry.meta.k_size); }
+
+  StringView Value() const {
+    return StringView(data + entry.meta.k_size, entry.meta.v_size);
+  }
+
+  void PersistNextNT(MemoryOffsetType offset) {
```
Comment (iyupeng) [inline volatile/engine/data_record.hpp:275]: Do we still need these functions in volatile module?

**Review by iyupeng — APPROVED** [review]

**Pull request merged**